src/clr-angular/data/datagrid/datagrid-detail.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+  Component,
+  Directive,
+  EventEmitter,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  Output,
+  OnInit,
+  OnDestroy,
+  ViewChild,
+  HostListener,
+} from '@angular/core';
+import { DetailService } from './providers/detail.service';
+import { Subscription } from 'rxjs';
+import { FocusTrapDirective } from '../../utils/focus-trap/focus-trap.directive';
+
+@Component({
+  selector: 'clr-dg-detail',
+  host: {
+    '[class.datagrid-detail-pane-wrapper]': 'true',
+  },
+  template: `
+    <div clrFocusTrap class="datagrid-detail-pane">
```

Comment:
If `clr-dg-detail` is placed inside the datagrid, it will be always present there. And we shouldn't have the focus trap a component that's always present.

Reply:
Good call, I will have to put an *ngIf on this element.